### PR TITLE
fix molstogridimage on certain fragments/smarts patterns

### PR DESCRIPTION
**Pull request by bp-kelley**
rdkit.Chem.Draw was failing on MolsToImage and MolsToGridImage

```
from rdkit.Chem import FilterCatalog
fgroups = FilterCatalog.GetFlattenedFunctionalGroupHierarchy()
Draw.MolsToImage(pats)
```

Resulted in

```
ArgumentError: Python argument types in
    MolDraw2D.DrawMolecule(MolDraw2DCairo, Mol)
did not match C++ signature:
    DrawMolecule(RDKit::MolDraw2D {lvalue} self, RDKit::ROMol mol, boost::python::api::object highlightAtoms, boost::python::api::object highlightBonds, boost::python::api::object highlightAtomColors=None, boost::python::api::object highlightBondColors=None, boost::python::api::object highlightAtomRadii=None, int confId=-1, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > legend='')
    DrawMolecule(RDKit::MolDraw2D {lvalue} self, RDKit::ROMol mol, boost::python::api::object highlightAtoms=None, boost::python::api::object highlightAtomColors=None, boost::python::api::object highlightAtomRadii=None, int confId=-1, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > legend='')
```

```
Draw.MolsToGridImage(pats)
```

Resulted in 

```
RuntimeError: Pre-condition Violation
	getExplicitValence() called without call to calcExplicitValence()
	Violation occurred on line 196 in file Code/GraphMol/Atom.cpp
	Failed Expression: d_explicitValence > -1
	RDKIT: 2020.03.1
	BOOST: 1_67
```

The first commit fixes this, but then the jupyter notebook cell is filled with RDKit warnings/errors.

The second commit enables a temporary block of rdkit logging (this can be a separate commit if necessary).

### CODE REVIEW

**Comment by bp-kelley** [comment]
I found another kekulization issue in the notebook:

```
AllChem.ReactionFromSmarts("[c;H1:3]1:[c:4]:[c:5]:[c;H1:6]:[c:7]2:[nH:8]:[c:9]:[c;H1:1]:[c:2]:1:2.O=[C:10]1[#6;H2:11][#6;H2:12][N:13][#6;H2:14][#6;H2:15]1>>[#6;H2:12]3[#6;H1:11]=[C:10]([c:1]1:[c:9]:[n:8]:[c:7]2:[c:6]:[c:5]:[c:4]:[c:3]:[c:2]:1:2)[#6;H2:15][#6;H2:14][N:13]3")
```

```
KekulizeException: Can't kekulize mol.  Unkekulized atoms: 23 24 25 26 27
```

Working on it.

**Comment by bp-kelley** [comment]
@greglandrum This went down a bit of a rabbit hole, happy to split it up, but I think I resolved the issues I have found.

File: Code/RDGeneral/RDLog.h [hunk 1]
```diff
@@ -90,18 +92,14 @@ RDKIT_RDGENERAL_EXPORT std::ostream &toStream(std::ostream &);
...
-    rdDebugLog;
-RDKIT_RDGENERAL_EXPORT extern std::shared_ptr<boost::logging::rdLogger>
-    rdInfoLog;
-RDKIT_RDGENERAL_EXPORT extern std::shared_ptr<boost::logging::rdLogger>
-    rdErrorLog;
-RDKIT_RDGENERAL_EXPORT extern std::shared_ptr<boost::logging::rdLogger>
-    rdWarningLog;
-RDKIT_RDGENERAL_EXPORT extern std::shared_ptr<boost::logging::rdLogger>
-    rdStatusLog;
+typedef std::shared_ptr<boost::logging::rdLogger> RDLogger;
```
Comment (greglandrum) [inline Code/RDGeneral/RDLog.h:95]: I think we ought to switch to using `using` instead of `typedef` when we write new code
I will at some point do a `clang-tidy` pass across the code base to take care of the legacy, but we might as well not make that legacy larger. ;-)

File: Code/GraphMol/MolDraw2D/MolDraw2D.cpp [hunk 2]
```diff
@@ -327,9 +329,10 @@ void MolDraw2D::get2DCoordsMol(RWMol &mol, double &offset, double spacing,
   }
 
   const bool canonOrient = true;
+  const bool kekulize = false;
   RDDepict::compute2DCoords(mol, nullptr, canonOrient);
   MolDraw2DUtils::prepareMolForDrawing(
-      mol, false);  // don't kekulize, we just did that
+      mol, kekulize);  // don't kekulize, we just did that
```
Comment (greglandrum) [inline Code/GraphMol/MolDraw2D/MolDraw2D.cpp:335]: the comment was there for the naked bool. If you're going to name it, then please move the comment up to the definition of `kekulize`

File: Code/RDGeneral/RDLog.h [hunk 3]
```diff
@@ -115,5 +113,12 @@ BOOST_DECLARE_LOG(rdStatusLog)
 #endif
 namespace RDLog {
 RDKIT_RDGENERAL_EXPORT void InitLogs();
+
+// ! Temporarily block logging until this object goes out of scope
+struct RDKIT_RDGENERAL_EXPORT BlockLogs {
```
Comment (greglandrum) [inline Code/RDGeneral/RDLog.h:118]: ooooo, this is useful.

File: rdkit/Chem/Draw/UnitTestDraw.py [hunk 4]
```diff
@@ -41,7 +42,7 @@
 class TestCase(unittest.TestCase):
   showAllImages = False
 
-  def test_interactive(self):
+  def atest_interactive(self):
```
Comment (greglandrum) [inline rdkit/Chem/Draw/UnitTestDraw.py:45]: you disabled a bunch of tests in here by renaming them.
We can think about removing the tests of the old drawing code, but let's do that in separate PR if we go that way.
For this one please re-enable these tests.

Comment (bp-kelley) [reply rdkit/Chem/Draw/UnitTestDraw.py:45]: Arg!  I need a commit hook for this, I rename tests so I don’t go insane with logs :)

File: Code/RDGeneral/RDLog.h (hunk 1 above)
Comment (bp-kelley) [reply Code/RDGeneral/RDLog.h]: @greglandrum I'm not quite sure what you mean by "linked to the issue" do you want me to put in an rdkit issue and link to that?

Comment (greglandrum) [reply Code/RDGeneral/RDLog.h]: @bp-kelley : isn't this connected to #3218?

**Comment by greglandrum** [comment]
@bp-kelley : what's your take on which milestone this should be added to. Normally for a bug fix it'd be a no-brainer to do the 2020.03.4 release, but given the new logging features maybe this is 2020.09.1?
